src/code.rs

```diff
@@ -470,10 +507,56 @@ impl {struct_name} {{
     ));
 
     buffer.push_str(
-        r##"
-}"##,
+        r#"
+}"#,
     );
 
+    // generate filter struct for filter() helper function
+    {
+        let filter_fields = table
+            .columns
+            .iter()
+            .map(|column| {
+                let column_name = column.name.to_string();
+                format!(
+                    "pub {column_name}: Option<{column_type}>,",
+                    column_name = column_name,
+                    column_type = if column.is_nullable {
+                        format!("Option<{}>", column.ty)
+                    } else {
+                        column.ty.clone()
+                    }
+                )
+            })
+            .collect::<Vec<_>>()
+            .join("\n    ");
+        let filter_fields_default = table
+            .columns
+            .iter()
+            .map(|column| {
+                let column_name = column.name.to_string();
+                format!("{column_name}: None,")
+            })
+            .collect::<Vec<_>>()
+            .join("\n            ");
+        buffer.push_str(&format!(
+            r##"
+#[derive(Clone)]
```

Comment:
```suggestion
#[derive(Debug, Clone)]
```
the debug derive should always be there